From f134e7467920988a9259f06de60327c75613d782 Mon Sep 17 00:00:00 2001
From: Ruslan Bay <ruslanbey@proton.me>
Date: Thu, 26 Feb 2026 17:21:27 +0100
Subject: [PATCH 35/54] media: intel/ipu4: Adapt to v4l2_async_nf_register()
 API signature change

Update async notifier registration call to use the simplified v4l2_async_nf_register()
API that no longer requires the v4l2_device parameter (obtained from notifier context).
---
 drivers/media/pci/intel/ipu-isys.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/pci/intel/ipu-isys.c b/drivers/media/pci/intel/ipu-isys.c
index fec4dfb0b27b..75085e5d475f 100644
--- a/drivers/media/pci/intel/ipu-isys.c
+++ b/drivers/media/pci/intel/ipu-isys.c
@@ -492,7 +492,7 @@ static int isys_notifier_init(struct ipu_isys *isys)
 	}
 
 	isys->notifier.ops = &isys_async_ops;
-	ret = v4l2_async_nf_register(&isys->v4l2_dev, &isys->notifier);
+	ret = v4l2_async_nf_register(&isys->notifier);
 	if (ret) {
 		dev_err(dev, "failed to register async notifier : %d\n", ret);
 		v4l2_async_nf_cleanup(&isys->notifier);
-- 
2.51.0

